pages/api/books/index.js

```diff
@@ -1,43 +1,49 @@
-import Book from '../../../models/Book';
-import dbConnect from '../../../utils/dbConnect';
-import axios from 'axios';
+import Book from '../../../models/Book'
+import dbConnect from '../../../utils/dbConnect'
+import axios from '../../../utils/axiosClient'
 
-export default async(req,res) => {
-  const { method } = req;
-  
-  // connect to database
-  await dbConnect();
+const handler = async (req, res) => {
+	const { method } = req
 
-  // create book
-  if (method === "POST") {
-    try {
-      const newBook = await new Book(req.body).save();
-      res.status(201)
-        .json({ data: newBook, message: "Book added successfully!"});
-    } catch (error) {
-      res.status(500).json({ message: "internal server error"});
-      console.log(error);
-    }
-  }
+	// connect to database
+	await dbConnect()
 
-  if (method === "GET") {
-    try {
-      const url = `https://api.nytimes.com/svc/books/v3/lists//hardcover-fiction?api-key=${process.env.NYT_API}`
-      const booksData = await axios.get(url)
-      const books = booksData.data.results.books
-      // process results w/ the DB
-      for(let i=0; i<books.length; i++) {
-        const book = books[i];
-        const inDB = await Book.find( 
-          { isbn: String(book['primary_isbn13'])});
-        if (inDB.length > 0) {
-          book.added = true;
-        }
-      }
-      res.status(200).json({ data: books })
-    } catch (error) {
-      res.status(500).json({ message: "internal server error"});
-      console.log(error);
-    }
-  }
+	// create book
+	if (method === 'POST') {
+		try {
+			// const newBook = await new Book( req.body ).save()
+			const newBook = await Book.findOneAndUpdate(
+				{ isbn: req.body.isbn },
+				{ book: req.body.book },
+				{ new: true, upsert: true }
+			)
```

Comment:
When I was testing, i noticed a record got added twice when I clicked
on the heart. Upsert will make sure that there will only be one
entry per ISBN.